redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -60,35 +62,39 @@ trait Interpreter {
         Promise.make[RedisError, String].flatMap(p => pendingRequests.offer(Request(command, p)).as(p))
 
       private val executeNext: UIO[Any] =
-        pendingRequests.take.flatMap { request =>
+        pendingRequests.take.map { request =>
           val exchange =
-            IO {
-              unsafeSend(request.command)
-              unsafeReceive()
+            IO.effect {
+              try {
+                unsafeSend(request.command)
+                unsafeReceive()
+              } catch {
+                case t: Throwable => throw RedisError.ProtocolError(t.getMessage)
+              } finally readinessFlag.set(true)
             }
 
-          val completer = exchange.catchAll(RedisError.make).ensuring(UIO(readinessFlag.set(true)))
-
-          request.result.completeWith(completer)
+          request.result.unsafeDone(exchange.refineToOrDie[RedisError])
         }
 
       private def unsafeReceive(): String = {
-        val sb        = new StringBuilder
+        val cb        = ChunkBuilder.make[Byte]()
```

Comment:
Actually, I suggest: `ArrayBuilder[Chunk[Byte]]` (or maybe `ArrayBuilder[Array[Byte]]`. The `++=` on `ChunkBuilder` will end up boxing `Byte` (I think).